core/src/sigverify.rs

```diff
@@ -81,24 +133,55 @@ fn batch_size(batches: &[Packets]) -> usize {
     batches.iter().map(|p| p.packets.len()).sum()
 }
 
-pub fn get_packet_offsets(packet: &Packet, current_offset: u32) -> (u32, u32, u32, u32) {
-    let (sig_len, sig_size) = decode_len(&packet.data);
-    let msg_start_offset = sig_size + sig_len * size_of::<Signature>();
+// internal function to be unit-tested; should be used only by get_packet_offsets
+fn do_get_packet_offsets(packet: &Packet, current_offset: u32) -> UnsanitizedPacketOffsets {
+    // This directly reads the length of Transaction.signatures (serialized with short_vec)
+    let (sig_len_untrusted, sig_size) = decode_len(&packet.data);
+
+    // This directly reads MessageHeader.num_required_signatures (serialized with u8)
+    let msg_start_offset = sig_size + sig_len_untrusted * size_of::<Signature>();
+    // Using msg_start_offset which is based on sig_len_untrusted introduces uncertainty.
+    // Ultimately, the actual sigverify will determine the uncertainty.
+    let sig_len_maybe_trusted = packet.data[msg_start_offset] as usize;
+
     let msg_header_size = serialized_size(&MessageHeader::default()).unwrap() as usize;
 
-    let (_pubkey_len, pubkey_size) =
+    // This directly reads the length of Message.account_keys (serialized with short_vec)
+    let (_pubkey_len, pubkey_len_size) =
         decode_len(&packet.data[(msg_start_offset + msg_header_size)..]);
 
     let sig_start = current_offset as usize + sig_size;
     let msg_start = current_offset as usize + msg_start_offset;
-    let pubkey_start = msg_start + msg_header_size + pubkey_size;
-
-    (
-        sig_len as u32,
-        sig_start as u32,
-        msg_start as u32,
-        pubkey_start as u32,
-    )
+    let pubkey_start = msg_start + msg_header_size + pubkey_len_size;
+
+    if sig_len_maybe_trusted == sig_len_untrusted {
+        UnsanitizedPacketOffsets::new(
+            true,
+            sig_len_maybe_trusted as u32,
+            sig_start as u32,
+            msg_start as u32,
+            pubkey_start as u32,
+        )
+    } else {
+        // a malformed packet is detected!!
+        UnsanitizedPacketOffsets::new(
+            false,
+            sig_len_untrusted as u32,
+            sig_start as u32,
+            msg_start as u32,
+            pubkey_start as u32,
+        )
+    }
```

Comment:
Can simplify this section to:
```
UnsanitizedPacketOffsets::new(
     sig_len_maybe_trusted == sig_len_untrusted,
     sig_len_maybe_trusted as u32,
     sig_start as u32,
     msg_start as u32,
     pubkey_start as u32,
)

Reply:
done at 81e022a4ceb6517c055c6a051f6a8be5234291bc!